src/Cli/dotnet/commands/dotnet-new/SdkInfoProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.MSBuildSdkResolver;
+using Microsoft.DotNet.NativeWrapper;
+using Microsoft.TemplateEngine.Abstractions.Components;
+using Microsoft.TemplateEngine.Utils;
+
+namespace Microsoft.DotNet.Tools.New
+{
+    internal class SdkInfoProvider : ISdkInfoProvider
+    {
+        public Guid Id { get; } = Guid.Parse("{A846C4E2-1E85-4BF5-954D-17655D916928}");
```

Comment:
What's the purpose of the GUID?

Reply:
This is required by Template engine component architecture - components must have their guids (dictated by the [`IIdentifiedComponent` interface](https://github.com/dotnet/templating/blob/main/src/Microsoft.TemplateEngine.Abstractions/IIdentifiedComponent.cs)) - so that API users can inject/invoke them by guids ([doc](https://github.com/dotnet/templating/blob/main/docs/api/Inside-the-Template-Engine.md#components))